File: 0349-intersection-of-two-arrays/intersection-of-two-arrays.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number[]}
 */
var intersection = function(nums1, nums2) {
    let set = new Set(nums1);
    let res = [];

    for (let n of nums2) {
        if (set.delete(n)) {
            res.push(n);
        }
    }
    return res;
};

File: 0349-intersection-of-two-arrays/intersection-of-two-arrays.py
class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        neset = set(nums1)
        res = []

        for n in nums2:
            if n in neset: 
                res.append(n)
                neset.discard(n)
    
        return res